File: soyeong/programmers-2/행렬의-곱셈.py
def solution(arr1, arr2):
    # 행렬의 곱셈 개념을 알아야 함
    # 앞배열의 한 행을 뒷배열의 열에 곱함
    # [[A11,A12],[A21,A22],[A31,A32]] 행3 * [[B11,B12],[B21,B22]] 열2
    # = [[(A11*B11)+(A12*B21), (A11*B12)+(A12*B22)],
    #    [(A21*B11)+(A22*B21), (A21*B12)+(A22*B22)],
    #    [(A31*B11)+(A32*B21), (A31*B12)+(A32*B22)]]
    answer = []

    for i in range(len(arr1)):  # arr1 행
        answer1 = []
        for j in range(len(arr2[0])):  # arr2 열
            num = 0

            for k in range(len(arr1[0])):  # arr1 열
                num += arr1[i][k] * arr2[k][j]
            answer1.append(num)

        answer.append(answer1)

    return answer

File: ingwang/programmers-2/행렬의 곱셈.java
import java.util.ArrayList;

class Solution17 {
    public static int[][] solution17(int[][] arr1, int[][] arr2) {
        int[][] answer = new int[arr1.length][arr2[0].length];

        for (int i = 0; i < arr1.length; i++) {
            for (int j = 0; j < arr2[0].length; j++) {
                for (int j2 = 0; j2 < arr1[0].length; j2++) {
                    answer[i][j] +=arr1[i][j2] * arr2[j2][j];
                }
                
            }
        }
        return answer;
    }

    public static void main(String[] args) {
        int[][] a = {};
        int[][] b = {};
        System.out.println(solution17(a, b));
    }
}